compositor_pipeline/src/pipeline/input/decklink/find_device.rs

```diff
@@ -0,0 +1,49 @@
+use decklink::{
+    get_decklinks, FlagAttributeId, IntegerAttributeId, StringAttributeId, VideoIOSupport,
+};
+
+use super::{DeckLinkError, DeckLinkOptions};
+
+pub(super) fn find_decklink(opts: &DeckLinkOptions) -> Result<decklink::DeckLink, DeckLinkError> {
+    let decklinks = get_decklinks()?;
+
+    for mut decklink in decklinks.into_iter() {
+        if is_selected_decklink(opts, &mut decklink)? {
+            return Ok(decklink);
+        }
+    }
+    Err(DeckLinkError::NoMatchingDeckLink)
+}
+
+fn is_selected_decklink(
```

Comment:
this would make sense for validation, but this function is mostly checking if the current device is the one we want to use. I'll leave it for now, but keep in mind refactor it in the future into 2 separate function (one for selecting and one for checks)